### PR TITLE
Reliability improvements

### PR DESCRIPTION
**Pull request by barafm**
**🔧 Major StdPipe Enhancements (Critical Improvement):**
- Robust partial write handling - Now supports multi-GB data transfers with automatic retry logic
- Fixed Write() method edge cases - Prevents data loss during large transfers
- Added comprehensive 1MB stress testing - Validates reliability under heavy load
- Performance fix - Corrected incorrect std::move usage in Read() method

**🛡️ Security & Safety Upgrades:**
- Null pointer validation - Added comprehensive parameter checking with proper error codes
- RAII handle management - UniqueHandle wrapper prevents resource leaks
- Input validation - Enhanced safety with security documentation
- Exception-free error handling - Maintains stability while improving safety

**⚡ Performance & Code Quality:**
- Simplified argument handling - Reduced complexity from std::vector to std::string
- Defensive programming - Added early returns and edge case protection
- Enhanced documentation - Security guidance and API safety information

**🧪 Testing Excellence:**
- 49 comprehensive test cases with 136 assertions
- Stress testing with multi-megabyte data validation
- Modern doctest framework - Superior error reporting and debugging
- Binary data handling - Enhanced support for special characters and null bytes

(Used an AI buddy to inspect the code and made improvements regarding on the findings. This was actually a nice AI learning experience 🧠. )

### CODE REVIEW

File: PipedProcess/PipedProcess.h [hunk 1]
```diff
@@ -99,21 +103,31 @@ class PipedProcess
...
+        }
 
         try
         {
+            // arguments need to be in a writable buffer for the API call
+            std::string args = arguments;
+
             // Note: Raymond Chen ("The Old New Thing") has some thoughtful insights about pipes:
-			// "Be careful when redirecting both a process�s stdin and stdout to pipes, for you can easily deadlock"
+			// "Be careful when redirecting both a process�s stdin and stdout to pipes, for you can easily deadlock"
```
Comment (fmuecke) [inline PipedProcess/PipedProcess.h]:
```suggestion
			// "Be careful when redirecting both a process's stdin and stdout to pipes, for you can easily deadlock"

```

**Review by fmuecke — COMMENTED** [review]
a small typo
